### PR TITLE
Shops and Battle Areas

### PR DESCRIPTION
**Pull request by ethteck**
* Battle area data disassembled
* Some battle area funcs deduplicated
* Some shop funcs
* Misc cleanup

### CODE REVIEW

File: include/functions.h [hunk 1]
```diff
@@ -47,6 +47,11 @@ void func_802666E4(Actor* actor, f32 arg1, f32 arg2, f32 arg3, s16);
 
 void step_game_loop(void);
 s32 resume_all_group(s32 groupFlags);
+f32 length2D(f32 x, f32 y);
+
+void transform_point(Matrix4f mtx, f32, f32, f32, f32, f32*, f32*, f32*, f32*);
```
Comment (nanaian) [inline include/functions.h:52]: Gives these args names? e.g. x y z w outX outY outZ outW

File: src/common/BattleAreaAngleStuff.inc.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    return ret;
+}
+
+ApiStatus N(BattleAreaAngleStuff2)(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode *args = script->ptrReadPos;
+    s32 var1 = get_variable(script, *args++);
+    s32 var2 = get_variable(script, *args++);
+    s32 var3 = get_variable(script, *args++);
+    s32 var4 = get_variable(script, *args++);
+    s32 var5 = get_variable(script, *args);
```
Comment (nanaian) [inline src/common/BattleAreaAngleStuff.inc.c:49]: *args++

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/common/BattleAreaAngleStuff.inc.c (hunk 2 above)
Comment (ethteck) [reply src/common/BattleAreaAngleStuff.inc.c:49]: It's not, and that interferes with the future uses of it

Comment (nanaian) [reply src/common/BattleAreaAngleStuff.inc.c:49]: oh, i didn't realize there were future uses. whoops!

Comment (ethteck) [reply src/common/BattleAreaAngleStuff.inc.c:49]: no problem, boblem

**Pull request merged**